docs/content/testing/testing-guidelines.md

```diff
@@ -0,0 +1,238 @@
+- `<ClassName>StaticTests`
+- `<ClassName>InstanceTests`
+
+The `StaticTests` may not need to set anything up, but the `InstanceTests` may need to create an instance of the class to be able to verify the instance methods.
+
+There are other reasons to create multiple fixtures, for example if you would like to test different pre-conditions.
+
+In most cases, each test file will contain one fixture. In this case, this fixture can be named `<ClassName>Tests`.
+
+The test methods should match this pattern:
+
+> `<method>_should_<action>_when_<conditions>`.
+
+### Setup and teardown
+
+The following snippet shows the recommended order of the NUnit set up and tear
+ down annotations:
+
+```cs
+[TestFixture]
+public class ExampleTests
+{
+    // Proposed structure:
+    // - one time set up and tear down
+    // - per test set up and tear down
+    // - each test one after another
+
+    [OneTimeSetUp]
+    public void OneTimeSetup()
+    {
+        // Set up resources, variables and parameters that may not change per test
+        // E.g. load a scene that all tests in this fixture will depend on
+    }
+
+    [OneTimeTearDown]
+    public void OneTimeTearDown()
+    {
+        // Clean up the resources, variables and parameters that may not change per test
+        // E.g. unload the scene
+    }
+
+    [SetUp]
+    public void Setup()
+    {
+        // Set up resources, variables and parameters that may change within each test
+        // E.g. create a gameobject
+    }
+
+    [TearDown]
+    public void Teardown()
+    {
+        // Clean up the resources, variables and parameters that may change within each test
+        // E.g. delete the created gameobject
+    }
+
+    [Test]
+    public void X_should_Y_when_Z()
+    {
+        // Prepare case, call X, assert that Y happens
+    }
+
+    [Test]
+    public void P_should_Q_when_R()
+    {
+        // Prepare case, call P, assert that Q happens
+    }
+}
+```
+
+### Assertions
+
+For NUnit, you can use the `Assert` class to make assertions.
+
+Please pay attention to parameter names, for example, for `Assert.AreEqual`,
+ the first parameter is the expected value, and the second parameter is the
+ value we are testing.
+
+```cs
+  var sumResult = Add(2, 3);
+  Assert.AreEqual(5, sumResult);
+```
+
+#### Assertions for exceptions
+
+By default, tests will fail if any code within throws an exception.
+
+Sometimes you may want to test that an exception will be thrown in certain
+ situations.
+
+You can do this using the `Assert.Throws` method. This method returns the caught
+ exception, and you can do further assertions against that.
+
+```cs
+        [Test]
+        public void Validate_should_return_false_when_LocatorHost_is_empty()
+        {
+            var config = GetDefaultWorkingConfig();
+            config.LocatorHost = "";
+
+            var exception = Assert.Throws<System.ArgumentException>(() => config.Validate());
```

Comment:
nit: you could simplify this to just `var exception = Assert.Throws<System.ArgumentException>(config.Validate);`

Reply:
Neat :)